browser/src/Services/Explorer/ExplorerSplit.tsx

```diff
@@ -212,26 +261,85 @@ export class ExplorerSplit {
+    // expands they are not toggles
+    private _toggleDirectory(action: "expand" | "collapse"): void {
         const selectedItem = this._getSelectedItem()
+        if (!selectedItem || selectedItem.type !== "folder") {
+            return
+        }
+
+        const type = action === "expand" ? "EXPAND_DIRECTORY" : "COLLAPSE_DIRECTORY"
+        this._store.dispatch({ type, directoryPath: selectedItem.folderPath })
+    }
+
+    private _onUndoItem(): void {
+        const { register: { undo } } = this._store.getState()
+        if (undo.length) {
+            this._store.dispatch({ type: "UNDO" })
+            this._store.dispatch({ type: "REFRESH" })
+        }
+    }
 
+    private _onYankItem(): void {
+        const selectedItem = this._getSelectedItem()
         if (!selectedItem) {
             return
         }
 
-        switch (selectedItem.type) {
-            case "file":
-                rm(selectedItem.filePath)
-                break
-            case "folder":
-                rm("-rf", selectedItem.folderPath)
-                break
-            default:
-                alert("Not implemented yet")
+        const { register: { yank } } = this._store.getState()
+        const inYankRegister = yank.some(({ id }) => id === selectedItem.id)
+
+        if (!inYankRegister) {
+            this._store.dispatch({ type: "YANK", target: selectedItem })
+            this._store.dispatch({ type: "REFRESH" })
+        } else {
+            this._store.dispatch({ type: "CLEAR_REGISTER", id: selectedItem.id })
+        }
+    }
+
+    private _onPasteItem(): void {
+        const pasteTarget = this._getSelectedItem()
+        if (!pasteTarget) {
+            return
+        }
+
+        const { register: { yank } } = this._store.getState()
+
+        if (yank.length && pasteTarget) {
+            this._store.dispatch({ type: "PASTE", target: pasteTarget, pasted: yank })
+            yank.forEach(yankedItem => {
+                this.moveFileOrFolder(yankedItem, pasteTarget)
+                this._store.dispatch({ type: "CLEAR_REGISTER", id: yankedItem.id })
+            })
         }
+    }
 
-        this._store.dispatch({
-            type: "REFRESH",
+    private _onDeletePersistItem(): void {
+        const selectedItem = this._getSelectedItem()
+
+        if (!selectedItem) {
+            return
+        }
+        this._store.dispatch({ type: "DELETE", target: selectedItem, persist: true })
+        this._sendDeletionNotification(selectedItem.name)
+    }
+
+    private _onDeleteItem(persist: boolean = true): void {
+        const selectedItem = this._getSelectedItem()
+
+        if (!selectedItem) {
+            return
+        }
+        this._store.dispatch({ type: "DELETE", target: selectedItem, persist: false })
+        this._sendDeletionNotification(selectedItem.name)
+    }
+
+    private _sendDeletionNotification(name: string): void {
+        this.sendExplorerNotification({
+            title: `${name} deleted`,
```

Comment:
This notification looked a bit awkward to me - with the tile saying like "File1.txt deleted" and "File1.txt removed" in the body. The other notifications have a more general title - like "File moved" and then the body would have specifics, like: "File test.txt was moved to E:/oni/test_directory successfully".

Perhaps we could change this to:
```
title: "File deleted",
details: `${name} was deleted successfully`
```